src/Microsoft.AspNetCore.Razor.Tools/CompilerHost.cs

```diff
@@ -26,13 +27,20 @@ public override ServerResponse Execute(ServerRequest request, CancellationToken
                     return new RejectedServerResponse();
                 }
 
-                var app = new Application(cancellationToken);
-                var commandArgs = parsed.args.ToArray();
+                using (var writer = new StringWriter())
+                {
+                    var app = new Application(cancellationToken)
+                    {
+                        Out = writer,
+                        Error = writer,
+                    };
+                    var commandArgs = parsed.args.ToArray();
 
-                var exitCode = app.Execute(commandArgs);
-                var output = app.Out.ToString() ?? string.Empty;
+                    var exitCode = app.Execute(commandArgs);
+                    var output = writer.ToString() ?? string.Empty;
```

Comment:
Would it be possible to avoid the stringwriter + tostring if the logging is turned off? This could be a significant amount of text in some cases and could cause LOH issues.

Reply:
Hmm not sure. I'll look into it. Would it still be an issue if we use a StreamWriter with a MemoryStream and read from it?